required_fields.node.js

```diff
@@ -19,29 +26,618 @@ function castBool(value) {
     }
 }
 
+nullOrUndefined = [null, undefined];
+
+mandatoryMessage = "Missing value for mandatory field";
+
+flatRequiredFields = {
+    assessment_date : mandatoryMessage,
+    blower_door_test : mandatoryMessage,
+    conditioned_floor_area : mandatoryMessage,
+    floor_to_ceiling_height : mandatoryMessage,
+    num_floor_above_grade : mandatoryMessage,
+    number_bedrooms : mandatoryMessage,
+    orientation : mandatoryMessage,
+    shape : mandatoryMessage,
+    year_built : mandatoryMessage,
+    hot_water_type : mandatoryMessage,
+    hvac_fraction_1 : mandatoryMessage,
+    heating_type_1 : mandatoryMessage,
+    heating_fuel_1 : mandatoryMessage,
+    cooling_type_1 : mandatoryMessage,
+    roof_type_1 : mandatoryMessage,
+    floor_area_1 : mandatoryMessage,
+    wall_construction_same : mandatoryMessage,
+    window_construction_same : mandatoryMessage
+};
+
 module.exports = function (homeValues) {
-    let mandatoryMessage = "Missing value for mandatory field";
+    // If we are given the new version of the home object, we need to validate the nested version instead
+    /* To handle the update where all building information is inside the `building_unit` property and then `building` inside
+    return homeValues.building_unit ? getNestedRequiredFields(homeValues.building_unit) : getFlatRequiredFields(homeValues);
+     */
+    return homeValues.building ? getNestedRequiredFields(homeValues) : getFlatRequiredFields(homeValues);
+}
+
+function setupAJV () {
+    ajv.addSchema(NestedBuildingSchema);
```

Comment:
So I installed locally through yarn, but on subsequent calls I get
> Error: schema with key or id &quot;https://github.com/NREL/hescore-hpxml/blob/master/hescorehpxml/schemas/hescore_json.schema.json&quot; already exists

My guess is that the ajv definition is global, and it's trying to reinstantiate an existing schema with the same key.  We probably need to can this setup method in favor of it going into an index or schema file, or at the top level/build of this file.

Reply:
Makes sense. I want to say I had it there in case we needed to pass something in, but that no loner is the case. Making that change.